release/models/bgp/openconfig-bgp-policy.yang

```diff
@@ -796,6 +805,324 @@ module openconfig-bgp-policy {
+                  config false;
+                  uses ext-community-color-config;
+              }
+          }
+        }
+      }
+  }
+
+grouping ext-community-asn-type-config {
+  description "Grouping for asn-type extended community config.";
+  leaf asn {
+      type oc-inet:as-number;
+      description "2 byte Autonomous system number.";
+  }
+
+  leaf value {
+      type uint32;
+      description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-asn4-type-config {
+  description "Grouping for asn4-type extended community config.";
+  leaf asn {
+      type oc-inet:as-number;
+      description "4 Byte autonomous system number.";
+  }
+
+  leaf value {
+      type uint16;
+      description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-ip-type-config {
+  description "Grouping for ip-type extended community config.";
+  leaf ip-address {
+    type oc-inet:ip-address;
+    description "IPv4 Address for the community.";
+  }
+
+  leaf value {
+    type uint16;
+    description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-link-bw-config {
+  description "Grouping for link bandwidth extended community config.";
+  leaf asn {
+    type oc-inet:as-number;
+    description "Autonomous system number.";
+  }
+
+  leaf bandwidth {
+    type oc-types:ieeefloat32;
+    description "Bandwidth in bytes per second.";
+  }
+}
+
+grouping ext-community-route-target-config {
+  description "Grouping for route-target extended community config.";
+  leaf asn {
+    type oc-inet:as-number;
+    description "Autonomous system number.";
+  }
+
+  leaf value {
+    type uint16;
+    description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-route-target-ip-config {
+  description "Grouping for route-target-ip extended community
+  config.";
+  leaf ipv4-address {
+    type oc-inet:ipv4-address;
+    description "IPv4 address.";
+  }
+
+  leaf value {
+    type uint16;
+    description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-route-origin-config {
+  description "Grouping for origin extended community config.";
+  leaf asn {
+    type oc-inet:as-number;
+    description "Autonomous system number.";
+  }
+
+  leaf value {
+    type uint16;
+    description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-route-origin-ip-config {
+  description "Grouping for route-origin-ip extended community config.";
+  leaf ipv4-address {
+    type oc-inet:ipv4-address;
+    description "IPv4 address.";
+  }
+
+  leaf value {
+    type uint16;
+    description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-color-config {
+  description "Grouping for color extended community config.";
+  leaf match-type {
+    type oc-bgp-types:bgp-color-match-type;
+    description "Specifies the match criteria between BGP NH and
+    SR policy endpoint";
+  }
+
+  leaf value {
```

Comment:
https://datatracker.ietf.org/doc/html/draft-ietf-idr-segment-routing-te-policy-26#name-color-extended-community covers changes that are happening to the color extended community.  What this will eventually push this model to is a need to have a second leaf that isn't solely "value", it will need to cover the flags and other things.

Reply:
Added `flags` enum below to cover draft-ietf-idr-segment-routing-te-policy-26